tests/components/ituran/test_config_flow.py

```diff
@@ -0,0 +1,382 @@
+"""Test the Ituran config flow."""
+
+from unittest.mock import patch
+
+from pyituran.exceptions import IturanApiError, IturanAuthError
+
+from homeassistant import config_entries
+from homeassistant.components.ituran.const import (
+    CONF_ID_OR_PASSPORT,
+    CONF_MOBILE_ID,
+    CONF_OTP,
+    CONF_PHONE_NUMBER,
+    DOMAIN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import MOCK_CONFIG_DATA
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with (
+        patch(
+            "homeassistant.components.ituran.config_flow.Ituran.is_authenticated",
+            return_value=False,
+        ),
+        patch(
+            "homeassistant.components.ituran.config_flow.Ituran.request_otp",
+        ),
+        patch(
+            "homeassistant.components.ituran.config_flow.Ituran.authenticate",
+        ),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_ID_OR_PASSPORT: MOCK_CONFIG_DATA[CONF_ID_OR_PASSPORT],
+                CONF_PHONE_NUMBER: MOCK_CONFIG_DATA[CONF_PHONE_NUMBER],
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "otp"
+        assert result["errors"] == {}
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_OTP: "123456",
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result["type"] == FlowResultType.CREATE_ENTRY
```

Comment:
```suggestion
        assert result["type"] is FlowResultType.CREATE_ENTRY
```
compare result types with `is`